typed-racket-lib/typed-racket/typecheck/tc-apply.rkt

```diff
@@ -83,9 +83,8 @@
            ;; Takes a possible substitution and computes
            ;; the substituted range type if it is not #f
            (define (finish substitution)
-             (begin0
```

Comment:
@samth This change was buggy. The `define-begin0-extraction` rule didn't account for when the surrounding `define` is a function definition instead of a variable definition.

Reply:
Fixed in jackfirth/resyntax#445.